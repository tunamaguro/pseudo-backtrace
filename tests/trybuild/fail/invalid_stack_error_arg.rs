use pseudo_backtrace::StackError;

#[derive(Debug, StackError)]
struct BadAttr {
    #[stack_error(foo)]
    source: std::io::Error,
    #[location]
    location: &'static core::panic::Location<'static>,
}

impl core::fmt::Display for BadAttr {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
        "bad".fmt(f)
    }
}
impl core::error::Error for BadAttr {}

fn main() {}
