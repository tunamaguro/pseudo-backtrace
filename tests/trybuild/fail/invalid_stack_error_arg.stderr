error: invalid `#[stack_error(foo)]` attribute. expected `std` or `stacked`
 --> tests/trybuild/fail/invalid_stack_error_arg.rs:5:19
  |
5 |     #[stack_error(foo)]
  |                   ^^^
